cudax/include/cuda/experimental/__hierarchy/hierarchy_dimensions.cuh

```diff
@@ -767,45 +792,44 @@ constexpr auto make_hierarchy_fragment(L1&& l1, Levels&&... ls) noexcept
  * @par
  */
 template <typename L1, typename... Levels>
-constexpr auto make_hierarchy(L1&& l1, Levels&&... ls) noexcept
+constexpr auto make_hierarchy(L1 l1, Levels... ls) noexcept
 {
-  return detail::make_hierarchy_fragment_reversable<false, thread_level>(
-    ::cuda::std::forward<L1>(l1), ::cuda::std::forward<Levels>(ls)...);
+  return detail::__make_hierarchy_fragment<thread_level>()(detail::__as_level(l1), detail::__as_level(ls)...);
 }
 
-// We can consider removing the operator, but its convinient for in-line construction
+// We can consider removing the operator&, but its convenient for in-line construction
 // TODO accept forwarding references
 template <typename LUnit, typename L1, typename... Levels>
-_CCCL_HOST_DEVICE constexpr auto
-operator&(const hierarchy_dimensions_fragment<LUnit, Levels...>& ls, const L1& l1) noexcept
+_CUDAX_API constexpr auto operator&(const hierarchy_dimensions_fragment<LUnit, Levels...>& ls, L1 l1) noexcept
```

Comment:
Nit: With the changes I noticed how bad L1/l1 name is here, its a copy paste from other places probably. Maybe `new_level` or similar would be better here?